src/ds5/ds5-options.cpp

```diff
@@ -502,18 +521,24 @@ namespace librealsense
 
     void alternating_emitter_option::set(float value)
     {
-        std::vector<uint8_t> pattern{};
+        if (_sensor && _sensor->supports_option(RS2_OPTION_HDR_ENABLED) && 
```

Comment:
The option internal logic should not change and the condition must be applied externally.
To be refactored so that the affected options, including LRS_ON LSR_POWER, EMITER_ON_OFF will be redirected via HDR block that will choose the action.
Also - there is a `gated_option` that partially addresses this.

Reply:
Done